File: chat-uikit/src/main/java/io/agora/chat/uikit/chathistory/EaseChatHistoryAdapter.java
package io.agora.chat.uikit.chathistory;

import android.view.ViewGroup;


import io.agora.chat.ChatMessage;
import io.agora.chat.uikit.chat.adapter.EaseMessageAdapter;
import io.agora.chat.uikit.chat.viewholder.EaseChatViewHolderFactory;
import io.agora.chat.uikit.chat.viewholder.EaseMessageViewType;

public class EaseChatHistoryAdapter extends EaseMessageAdapter {

    public EaseChatHistoryAdapter() {}

    @Override
    public int getItemNotEmptyViewType(int position) {
        return EaseChatViewHolderFactory.getViewType(mData.get(position));
    }

    @Override
    public ViewHolder<ChatMessage> getViewHolder(ViewGroup parent, int viewType) {
        return EaseChatViewHolderFactory.createHistoryViewHolder(parent, EaseMessageViewType.from(viewType));
    }

}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/feature/chat/chathistory/EaseChatHistoryAdapter.kt
package io.agora.uikit.feature.chat.chathistory

import android.view.ViewGroup
import io.agora.uikit.common.ChatMessage
import io.agora.uikit.feature.chat.adapter.EaseMessagesAdapter
import io.agora.uikit.feature.chat.chathistory.viewholder.EaseChatHistoryViewHolderFactory
import io.agora.uikit.feature.chat.viewholders.EaseMessageViewType

class EaseChatHistoryAdapter : EaseMessagesAdapter() {
    override fun getItemNotEmptyViewType(position: Int): Int {
        return EaseChatHistoryViewHolderFactory.getViewType(mData?.get(position))
    }

    override fun getViewHolder(parent: ViewGroup, viewType: Int): ViewHolder<ChatMessage> {
        return EaseChatHistoryViewHolderFactory.createViewHolder(
            parent,
            EaseMessageViewType.from(viewType)
        )
    }

}
